apps/prairielearn/assets/scripts/lib/editAccessRuleModal.ts

```diff
@@ -0,0 +1,170 @@
+import { html, escapeHtml } from '@prairielearn/html';
+
+import { Modal } from '../../../src/components/Modal.html.js';
+import { AssessmentAccessRules } from '../../../src/pages/instructorAssessmentAccess/instructorAssessmentAccess.types.js';
+
+export function EditAccessRuleModal({
+  accessRule,
+  i,
+  addAccessRule = false,
+  timeZoneName,
+}: {
+  accessRule: AssessmentAccessRules;
+  i: number;
+  addAccessRule?: boolean;
+  timeZoneName: string;
+}) {
+  return Modal({
+    id: `editAccessRuleModal`,
+    title: `${addAccessRule ? 'Add Access Rule' : `Edit Access Rule ${i + 1}`}`,
+    body: html`
+      <div class="form-group">
+        <label for="mode">Mode</label>
+        <small class="form-text text-muted">
+          Used to restrict access to assessments based on the current mode. In general, it is best
+          to use "Public" for any homework and "Exam" for exams in the Computer Based Testing
+          Facility.
+        </small>
+        <select class="form-control" id="mode" name="mode">
+          <option value="" ${accessRule.mode === '' ? html`selected` : ''}>—</option>
+          <option value="Exam" ${accessRule.mode === 'Exam' ? html`selected` : ''}>Exam</option>
+          <option value="Public" ${accessRule.mode === 'Public' ? html`selected` : ''}>
+            Public
+          </option>
+        </select>
+      </div>
+      <div class="form-group">
+        <label for="uids">UIDs</label>
+        <small class="form-text text-muted">
+          Require one of the UIDs in the array to access. Enter UIDs separated by commas or leave
+          blank to allow all enrolled students to access.
+        </small>
```

Comment:
Here and elsewhere, help text like this should appear below the input. See https://getbootstrap.com/docs/4.6/components/forms/#help-text.

You'll also want to take care to associate this text with the corresponding input using `aria-describedby`.

Reply:
Moved help text below inputs and added `aria-describedby` to the corresponding inputs in [1dfef07](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/1dfef07527ed0296ff3cdfa7c58d85fce2304b97).